### PR TITLE
Allow modules to disable routing prefix with specific parameter

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | When a module declares Symfony routes, these routes automatically get the prefix `/modules`. <br/>This can be an issue if module wants to re-map a Core route to its custom controller. I introduce a new `default` parameter for PrestaShop module routes that allow modules to disable routing prefix with specific parameter `_disable_module_prefix`
| Type?         | new feature
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | See below

## How to use it in a module

```
admin_orders_index:
  path: /sell/orders/orders/
  methods: [GET]
  defaults:
    _controller: Matks\TestDecoration\Controller\AController::aAction
    _legacy_controller: AdminOrders
    _legacy_link: AdminOrders
    _disable_module_prefix: true
```
=> if `_disable_module_prefix` is `true` the route is not prefixed

## How to test

Use following test module 
[testdecorationcontrollers.zip](https://github.com/PrestaShop/PrestaShop/files/4786683/testdecorationcontrollers.zip)

### On 1.7.7.x (without the PR)

Install the module on 1.7.7.x and look and click on Sell > Orders link. You arrive on a blank page with "ok". What is interesting is the URL: it should be `/admin-dev/index.php/modules/sell/orders/orders`
=> the module modified the URL of the "Orders" page 😢 

### On my branch (with the PR)

Install the module with my PR and look and click on Sell > Orders link. You arrive on a blank page with "ok". The URL is  `/admin-dev/index.php/sell/orders/orders`
=> fixed !

The snippet of code above allows a module developer to re-map the Orders page listing route `admin_orders_index` to its custom controller. Without the `_disable_module_prefix` option, the route is transformed by PrestaShop into `/modules/sell/orders/orders/`. So any link to the orders listing generated by PrestaShop will become `/modules/sell/orders/orders/` but external links and hardcoded links will fail.

With `_disable_module_prefix` set to true, the prefix is disabled and the route path remains `/sell/orders/orders/ ` which is the goal of this PR

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19782)

### CODE REVIEW

**Review by jolelievre — DISMISSED** [review]
Nice @matks

File: src/PrestaShopBundle/Routing/YamlModuleLoader.php [hunk 1]
```diff
@@ -82,4 +84,24 @@ public function supports($resource, $type = null)
...
+    /**
+     * @param RouteCollection $routes
+     *
+     * @return RouteCollection
+     */
+    private function modifyRoutes(RouteCollection $routes)
+    {
+        foreach ($routes->getIterator() as $route) {
+            if ($route->hasDefault('_disable_module_prefix')) {
+                if ($route->getDefault('_disable_module_prefix') === true) {
```
Comment (PululuK) [inline src/PrestaShopBundle/Routing/YamlModuleLoader.php]: Use `&&` operator instaed of double `if` statement.

**Review by matthieu-rolland — APPROVED** [review]

File: src/PrestaShopBundle/Routing/YamlModuleLoader.php (hunk 1 above)
Comment (matthieu-rolland) [reply src/PrestaShopBundle/Routing/YamlModuleLoader.php]: done

**Review by PierreRambaud — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

File: src/PrestaShopBundle/Routing/YamlModuleLoader.php (hunk 1 above)
Comment (matks) [reply src/PrestaShopBundle/Routing/YamlModuleLoader.php]: Indeed 🤔

Comment (matks) [reply src/PrestaShopBundle/Routing/YamlModuleLoader.php]: Thanks @matthieu-rolland for fixing my branch 😉

**Comment by matks** [comment]
Thanks for review ! The test module has been put in PR's body

**Comment by Progi1984** [comment]
Thanks @matks

**Comment by matks** [comment]
PR to update the devdocs accordingly https://github.com/PrestaShop/docs/pull/589

**Comment by Progi1984** [comment]
@matks PR Merged

**Pull request merged**